AppCenterLoaderApp/appcenter-loader/src/main/java/com/microsoft/appcenter/loader/AppCenterLoader.java

```diff
@@ -105,7 +106,14 @@ public boolean onCreate() {
         }
         if (isTrueValue(getStringResource(USE_PUSH_KEY)) &&
             isModuleAvailable("com.microsoft.appcenter.push.Push", "Push")) {
-            Push.setListener(new UnityAppCenterPushDelegate());
+            try {
+                @SuppressWarnings("unchecked")
+                Class<PushListener> pushClass = (Class<PushListener>) Class.forName("com.microsoft.appcenter.pushdelegate.UnityAppCenterPushDelegate");
+                Push.setListener(pushClass.newInstance());
+            } catch (Exception e) {
+                Log.i(TAG, "UnityAppCenterPushDelegate is not available: " + e.getMessage());
```

Comment:
If the check box is checked, the absence of push should be to the very least a warning.

And it should use AppCenterLog, and since its an exception, you can pass it as the third parameter.